### PR TITLE
Cleaned up and improved log messages

### PR DESCRIPTION
**Pull request by pboothe**
Cleaned up and improved log messages, and one improvement to the signal handling code to prevent it from leaking an open file.

### CODE REVIEW

File: src/logging.c [hunk 1]
```diff
@@ -367,17 +366,27 @@ void log_print(int lvl, const char* format, ...) {
...
  * @param lvl     level of the message
+ * @param file    filename where the log occurred
+ * @param line    line number in the file
+ * @param lvl     level of the message
  * @param format  format of the message
  *            ... - the additional arguments
  */
 
-void log_println(int lvl, const char* format, ...) {
+void log_println_impl(int lvl, const char* file, int line, const char* format, ...) {
```
Comment (stephen-soltesz) [inline src/logging.c]: I have a nagging feeling that these two functions could be combined or somehow simpler. But I cannot quantify it, which makes me thing that it probably wouldn't be quick. It's not apparent to me what additional value the I2ErrLogVT functions (or I2ErrHandles) provide here over direct implementation.

File: src/web100srv.c [hunk 2]
```diff
@@ -367,14 +367,24 @@ void cleanup(int signo) {
...
+          if (get_debuglvl() > 4) {
+            fprintf(fp, "Received SIGPIPE: a child has terminated early.\n");
+          }
+          fclose(fp);
+        }
+      } else {
+	// This is the SIGPIPE handler for a child. SIGPIPE should only
+	// happen if the client disconnected mid-test, and if the client does
+	// that, we should cause the child to exit.
+	log_println(1, "Child received SIGPIPE, exiting");
```
Comment (stephen-soltesz) [inline src/web100srv.c]: nit: looks like tabs?

**Comment by stephen-soltesz** [comment]
LGTM. Please add a TODO around log_println_impl to audit the calls to log_print*. This sends a public message that we intend to follow up.

File: src/logging.c (hunk 1 above)
Comment (pboothe) [inline src/logging.c]: I agree on both. I also don't know what value is provided by I2ErrLogVT, but tearing it out would be a lot of work and would likely break a few things in small ways.  IMO a long term goal of NDT should be to get rid of the I2Util libraries entirely.

File: src/web100srv.c (hunk 2 above)
Comment (pboothe) [inline src/web100srv.c]: Friggin vim. "set expandtab" should really be the default.

**Comment by pboothe** [comment]
Done!  Merging now.

**Pull request merged**